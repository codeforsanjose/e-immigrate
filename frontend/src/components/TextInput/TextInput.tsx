import React from 'react';
import './TextInput.css';
import { CommonComponentProps } from '../../types/CommonComponentProps';
type TextInputProps = CommonComponentProps;
export function TextInput(props: TextInputProps) {
    const {
        q, 
        bindField, 
        collectAnswer, 
        content,
    } = props;
    return (
        <>
            <input
                type="text"
                name={q.slug}
                required={q.required}
                className="TextInput"
                {...bindField(q.slug)}
                onChange={(e) => collectAnswer(q.slug, e.target.value)} />
            <div className="RequiredError">*{content.errorMessage}</div>
        </>
    );
}
