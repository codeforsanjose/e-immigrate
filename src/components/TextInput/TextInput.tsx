import React from 'react';
import './TextInput.css';
import { QData, BindFieldFunction, CollectAnswerFunction, ReactSetter } from '../../types/common';
type TextInputProps = {
    attributes: {
        q: QData;
        bindField: BindFieldFunction;
        collectAnswer: CollectAnswerFunction;
        content: {
            errorMessage: string;
        };
    }
};
export function TextInput(props: TextInputProps) {
    const {
        attributes: { 
            q, 
            bindField, 
            collectAnswer, 
            content,
        },
    } = props;
    return (
        <>
            <input
                type="text"
                name={q.slug}
                required={q.required}
                className="TextInput"
                {...bindField(q.slug)}
                onChange={(e) => collectAnswer(q.slug, e.target.value)} />
            <div className="RequiredError">*{content.errorMessage}</div>
        </>
    );
}
