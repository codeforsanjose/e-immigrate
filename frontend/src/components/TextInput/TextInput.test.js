import React from 'react';
import TextInput from './TextInput';

import renderer from 'react-test-renderer';

it('TextInput renders correctly', () => {
    const tree = renderer
        .create(
            <TextInput
                q={{}}
                answers={[]}
                values={[]}
                bindField={() => {}}
                showFollowUp={{}}
                setShowFollowUp={{}}
                collectAnswer={{}}
                content={{}}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
