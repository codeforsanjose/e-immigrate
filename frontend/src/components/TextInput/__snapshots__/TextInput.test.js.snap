// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
Array [
  <input
    className="TextInput"
    onChange={[Function]}
    type="text"
  />,
  <div
    className="RequiredError"
  >
    *
  </div>,
]
`;
